packages/messages/src/browser/notifications.ts

```diff
@@ -100,13 +123,63 @@ export class Notifications {
     }
 
     protected toIconClass(icon: string): string {
-        if (icon === 'error') {
-            return 'fa-times-circle';
+        switch (icon) {
+            case 'error': return 'fa-times-circle';
+            case 'warning': return 'fa-warning';
+            case 'progress': return 'fa-spinner';
+            default: return 'fa-info-circle';
+        }
+    }
+
+}
+
+class ProgressNotificationImpl implements ProgressNotification {
+    private increment: number = 0;
+    private readonly node: Node;
+    private readonly container: Element;
+    private readonly properties: NotificationProperties;
+
+    constructor(container: Element, node: Node, properties: NotificationProperties) {
+        this.node = node;
+        this.container = container;
+        this.properties = properties;
+    }
+
+    close(): void {
+        const element = document.getElementById('notification-container-' + this.properties.id);
+        if (!element) {
+            return;
         }
-        if (icon === 'warning') {
-            return 'fa-warning';
+        element.remove();
+    }
+
+    show(): void {
+        let container = document.getElementById('notification-container-' + this.properties.id);
+        if (!container) {
+            this.container.appendChild(this.node);
+        }
+        container = document.getElementById('notification-container-' + this.properties.id);
+        if (container) {
+            const progressContainer = container.appendChild(document.createElement('div'));
+            progressContainer.className = 'progress';
+            const progress = progressContainer.appendChild(document.createElement('p'));
+            progress.id = 'notification-progress-' + this.properties.id;
         }
-        return 'fa-info-circle';
     }
 
+    update(item: { message?: string, increment?: number }): void {
+        const textElement = document.getElementById('notification-text-' + this.properties.id);
+        if (textElement) {
+            if (item.increment) {
+                this.increment = this.increment + item.increment;
```

Comment:
this basically means that on window reload it starts with `0` again, right?

so the client should be responsible to preserve a state. 

what happens on reconnect actually? need to test it. 

have you thought about sending the actual percentage value? or maybe something like: `done` and `total` values?

Reply:
can we align with https://github.com/theia-ide/theia/pull/2461/files#diff-23694302557751638a1bd2f4e8c163b1R21 ?

Reply:
@vinokurig, could you please change the api and implementation to have `workDone` and `totalWork`. I think it's then good to merge (and we can use it for git clone reporting).

Reply:
I have a question: If the progress notification shows procent rate, why do we need `totalWork`? Can we use just `workDone` ?

Reply:
One can argue this might be more flexible. I've seen implementation of progress which started reporting e.g. with `2/4`, then the computation resolved the next task with 100 items, and reported `53/102` in the following update.

Reply:
Ok, thanks, so I'll change the notification view to show `<workDone>/<totalWork>` instead of instead of percent rate.

Reply:
@vinokurig, wait, in the presentation it still can be shown as percentage value.

Reply:
Do you mean that I need to keep percentage rate? but what if `totalWork !== 100`?

Reply:
`` `${Math.floor(work.done / work.total * 100)} %` `` should work, no?

Reply:
Now I got it, thank you